### PR TITLE
[DEV-6381] Preserve jira priorities and issue types order in charts a…

### PR DESCRIPTION
**Pull request by gaetano-guerriero**
…nd goals

- - -
This change is [Reviewable](https://reviewable.io/reviews/athenianco/athenian-api/3566)

### CODE REVIEW

File: server/athenian/api/internal/jira.py [hunk 1]
```diff
@@ -513,18 +513,20 @@ def normalize_user_type(type_: str) -> str:
...
 
 
-def parse_request_issue_types(req_issue_types: Optional[list[str]]) -> Optional[list[str]]:
+def parse_request_issue_types(req_issue_types: list[str] | None) -> list[str] | None:
     """Parse the raw Jira issue types received in a request."""
     if req_issue_types is None:
         return None
-    return sorted({normalize_issue_type(t) for t in req_issue_types})
+    normalized = [normalize_issue_type(t) for t in req_issue_types]
+    return list({t: None for t in normalized})
```
Comment (vmarkovtsev) [inline server/athenian/api/internal/jira.py:529]:
```suggestion
    return list({(normalize_issue_type(t) for t in req_issue_types})
```

Comment (gaetano-guerriero) [reply server/athenian/api/internal/jira.py:529]: set step will forget the req_issue_types order here

Comment (vmarkovtsev) [reply server/athenian/api/internal/jira.py:529]:
```python
return list({normalize_issue_type(t): None for t in req_issue_types})
```

Comment (vmarkovtsev) [reply server/athenian/api/internal/jira.py:529]: otherwise, we create a list, then a dict, then a list.

**Review by vmarkovtsev — APPROVED** [review]

**Pull request merged**